src/parse.ts

```diff
@@ -0,0 +1,137 @@
+import DataFactory from './factories/extended-term-factory'
+import jsonldParser from './jsonldparser'
+// @ts-ignore is this injected?
+import { Parser as N3jsParser } from 'n3'  // @@ Goal: remove this dependency
+import N3Parser from './n3parser'
+import { parseRDFaDOM } from './rdfaparser'
+import RDFParser from './rdfxmlparser'
+import sparqlUpdateParser from './patch-parser'
+import * as Util from './util'
+import Formula from './formula'
+import { ContentType } from './types'
+import { Quad } from './tf-types'
+
+type CallbackFunc = (error: any, kb: Formula | null) => void
+
+/**
+ * Parse a string and put the result into the graph kb.
+ * Normal method is sync.
+ * Unfortunately jsdonld is currently written to need to be called async.
+ * Hence the mess below with executeCallback.
+ * @param str - The input string to parse
+ * @param kb - The store to use
+ * @param base - The base URI to use
+ * @param contentType - The MIME content type string for the input
+ * @param callback - The callback to call when the data has been loaded
+ */
+export default function parse (
+  str: string,
+  kb: Formula,
+  base: string,
+  contentType: string | ContentType,
+  callback?: CallbackFunc
+) {
+  contentType = contentType || ContentType.turtle
+  contentType = contentType.split(';')[0] as ContentType
+  try {
+    if (contentType === ContentType.n3 || contentType === ContentType.turtle) {
+      var p = N3Parser(kb, kb, base, base, null, null, '', null)
+      p.loadBuf(str)
+      executeCallback()
+    } else if (contentType === ContentType.rdfxml) {
+      var parser = new RDFParser(kb)
+      parser.parse(Util.parseXML(str), base, kb.sym(base))
+      executeCallback()
+    } else if (contentType === ContentType.xhtml) {
+      parseRDFaDOM(Util.parseXML(str, {contentType: ContentType.xhtml}), kb, base)
+      executeCallback()
+    } else if (contentType === ContentType.html) {
+      parseRDFaDOM(Util.parseXML(str, {contentType: ContentType.html}), kb, base)
+      executeCallback()
+    } else if (contentType === ContentType.sparqlupdate) { // @@ we handle a subset
+      sparqlUpdateParser(str, kb, base)
+      executeCallback()
+    } else if (contentType === ContentType.jsonld) {
+      jsonldParser(str, kb, base, executeCallback)
+    } else if (contentType === ContentType.nQuads ||
+               contentType === ContentType.nQuadsAlt) {
+      var n3Parser = new N3jsParser({ factory: DataFactory })
+      nquadCallback(null, str)
+    } else if (contentType === undefined) {
+      throw new Error("contentType is undefined")
+    } else {
+      throw new Error("Don't know how to parse " + contentType + ' yet')
+    }
+  } catch (e) {
+    executeErrorCallback(e)
```

Comment:
That code is a bit weird, but it was already there. A comment above mentions that it has to do with JSON-LD parsing, maybe it can be removed? I'll ask Thom en Tim, but it will be a separate issue / PR.

Reply:
Yeah, if it was already there a separate PR makes more sense - or just don't bother, if it's not causing any problems now.